CoreScriptsRoot/CoreScripts/FriendPlayerPrompt.lua

```diff
@@ -0,0 +1,180 @@
+--[[
+	// Filename: FriendPlayerPrompt.lua
+	// Version 1.0
+	// Written by: TheGamer101
+	// Description: Can prompt a user to send a friend request or unfriend a player.
+]]--
+
+local StarterGui = game:GetService("StarterGui")
+local PlayersService = game:GetService("Players")
+local CoreGuiService = game:GetService("CoreGui")
+
+local RobloxGui = CoreGuiService:WaitForChild("RobloxGui")
+local LocalPlayer = PlayersService.LocalPlayer
+while LocalPlayer == nil do
+	PlayersService.ChildAdded:wait()
+	LocalPlayer = PlayersService.LocalPlayer
+end
+
+local CoreGuiModules = RobloxGui:WaitForChild("Modules")
+local PromptCreator = require(CoreGuiModules:WaitForChild("PromptCreator"))
+local PlayerDropDownModule = require(CoreGuiModules:WaitForChild("PlayerDropDown"))
+
+local readFlagSuccess, flagEnabled = pcall(function() return settings():GetFFlag("CorescriptPromptFriendEnabled") end)
+local AllowPromptFriendPlayer = readFlagSuccess and flagEnabled
+
+local THUMBNAIL_URL = "https://www.roblox.com/Thumbs/Avatar.ashx?x=200&y=200&userId="
+local BUST_THUMBNAIL_URL = "https://www.roblox.com/bust-thumbnail/image?width=420&height=420&userId="
+
+function SendFriendRequest(playerToFriend)
+	local success = pcall(function()
+		LocalPlayer:RequestFriendship(playerToFriend)
+	end)
+	return success
+end
+
+function AtFriendLimit()
+	local friendCount = PlayerDropDownModule:GetFriendCountAsync(LocalPlayer)
+	if friendCount >= PlayerDropDownModule:MaxFriendCount() then
+		return true
+	end
+	return false
+end
+
+function DoPromptRequestFriendPlayer(playerToFriend)
+	if LocalPlayer:IsFriendsWith(playerToFriend.UserId) then
```

Comment:
:IsFriendsWith uses getFriendStatus as it's first check internally. It does an additional check for the case where getFriendStatus would return Unknown.